src/probability/borel_cantelli.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2022 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import probability.martingale.borel_cantelli
+import probability.independence
+
+/-!
+
+# The second Borel-Cantelli lemma
+
+This file contains the second Borel-Cantelli lemma which states that, given a sequence of
+independent sets `(sₙ)` in a probability space, if `∑ n, μ sₙ = ∞`, then the limsup of `sₙ` has
+measure 1. We employ a proof using Lévy's generalized Borel-Cantelli by choosing an appropriate
+filtration.
+
+## Main result
+
+- `probability_theory.measure_limsup_eq_one`: the second Borel-Cantelli lemma.
+
+-/
+
+open_locale measure_theory probability_theory ennreal big_operators topological_space
+
+open measure_theory probability_theory measurable_space topological_space
+
+namespace probability_theory
+
+variables {Ω : Type*} {m0 : measurable_space Ω} {μ : measure Ω} {s : ℕ → set Ω}
+  [is_probability_measure μ]
+
+section borel_cantelli
+
+variables {β : Type*} [mβ : measurable_space β] [normed_add_comm_group β] [borel_space β]
+
+lemma Indep_fun.indep_comap_succ_natural {f : ℕ → Ω → β}
+  (hf : ∀ (i : ℕ), strongly_measurable (f i))
+  (hfi : Indep_fun (λ (n : ℕ), mβ) f μ) (n : ℕ) :
+  indep (measurable_space.comap (f (n + 1)) mβ) (filtration.natural f hf n) μ :=
```

Comment:
This can be generalized, instead of `n` and `n+1` in nat, to any `i ≤ j` in a more general type.
Could you check other lemmas for similar generalizations?